src/ImageSharp/Common/Helpers/Numerics.cs

```diff
@@ -970,6 +970,7 @@ public static uint RotateRightSoftwareFallback(uint value, int offset)
         /// <param name="value">Value.</param>
         /// <param name="min">Mininum value, inclusive.</param>
         /// <param name="max">Maximum value, inclusive.</param>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static bool IsOutOfRange(int value, int min, int max)
```

Comment:
This is a leftover from experiments on a similar helper method that deals with `(uint)x >= (uint)this.Width`, which made things slower actually. I'm afraid the helper method results in suboptimal code compared to keeping things inline.

/cc @gfoidl

Reply:
It depends if the helper is fed with constant arguments (like in the other PR with constant values for `min` and `max`) or if these values are non-constant, like here with `this.Height`.

> moves around a `bool` on the stack?

IMO this would be missed optimization from the JIT. A short method (which should be inlined by itself) + decorated with `AggressiveInlining` (so inlined for pretty sure), so the codegen should be equal.
But: sometimes -- so far I couldn't figure out when exactly -- the JIT won't optimize as expected when `this` is concerned, as there's some special checks for `this == null` which get sometimes removed, sometimes not. I expect this what is hit here.